### PR TITLE
CSHARP-4904: Add .NET 6 target framework

### PR DESCRIPTION
**Pull request by BorisDog**

### CODE REVIEW

File: src/MongoDB.Driver/Authentication/Oidc/FileOidcCallback.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@ public OidcAccessToken GetOidcAccessToken(OidcCallbackParameters parameters, Can
 
         public async Task<OidcAccessToken> GetOidcAccessTokenAsync(OidcCallbackParameters parameters, CancellationToken cancellationToken)
         {
-#if NETSTANDARD2_1_OR_GREATER
+#if NETSTANDARD2_1_OR_GREATER || NET6_0_OR_GREATER
```
Comment (sanych-sun) [inline src/MongoDB.Driver/Authentication/Oidc/FileOidcCallback.cs:48]: Do we really need this? Does `NETSTANDARD2_1_OR_GREATER` includes net6 as well?

File: src/MongoDB.Driver/Authentication/ScramSha/ScramCache.cs [hunk 2]
```diff
@@ -15,6 +15,7 @@
 
 using System.Linq;
 using System.Security;
+using MongoDB.Driver.Core.Misc;
```
Comment (sanych-sun) [inline src/MongoDB.Driver/Authentication/ScramSha/ScramCache.cs:18]: Why this using appeared?

File: tests/MongoDB.Driver.Tests/PasswordEvidenceTests.cs [hunk 3]
```diff
@@ -16,6 +16,7 @@
 using System;
 using System.Security;
 using FluentAssertions;
+using MongoDB.Driver.Core.Misc;
```
Comment (sanych-sun) [inline tests/MongoDB.Driver.Tests/PasswordEvidenceTests.cs:19]: Why this using appeared?

**Review by sanych-sun — CHANGES_REQUESTED** [review]

File: tests/MongoDB.Driver.Tests/PasswordEvidenceTests.cs (hunk 3 above)
Comment (BorisDog) [reply tests/MongoDB.Driver.Tests/PasswordEvidenceTests.cs:19]: We had two `DecryptedSecureString` classes.
Removed `MongoDB.Driver.DecryptedSecureString`, and left `MongoDB.Driver.Core.Misc.DecryptedSecureString`.

File: src/MongoDB.Driver/Authentication/ScramSha/ScramCache.cs (hunk 2 above)
Comment (BorisDog) [reply src/MongoDB.Driver/Authentication/ScramSha/ScramCache.cs:18]: We had two `DecryptedSecureString` classes.
Removed `MongoDB.Driver.DecryptedSecureString`, and left `MongoDB.Driver.Core.Misc.DecryptedSecureString`.

File: src/MongoDB.Driver/Authentication/Oidc/FileOidcCallback.cs (hunk 1 above)
Comment (BorisDog) [reply src/MongoDB.Driver/Authentication/Oidc/FileOidcCallback.cs:48]: `NETSTANDARD2_1_OR_GREATER` does not include `NET6_0_OR_GREATER`

**Review by sanych-sun — APPROVED** [review]
LGTM

**Pull request merged**